refactor(review): drop deterministic-analysis and autofix steps (#6092)
QwenLM/qwen-code: packages/core/src/skills/bundled/review/DESIGN.md

> - ❌ Adds two extra API calls (`check-runs` + `statuses`) per APPROVE-bound submit; only relevant for the `APPROVE` path so the cost is negligible.

Author: Fixed in `5172f9d8d` — section renamed to "Why auto-discover build/test commands from CI config …". 已改名。

> ## Why the deterministic checks live as `qwen review` subcommands

Author: **[Suggestion]** Section heading "Why the deterministic checks live as `qwen review` subcommands" is now misleading. After this PR, the section body only covers `presubmit` and `cleanup` subcommands — the linter/typecheck runner (`deterministic.ts`) is no longer part of the pipeline. Consider renaming to "Why presubmit and cleanup live as `qwen review` subcommands".

```suggestion
## Why presubmit and cleanup live as `qwen review` subcommands
```

_— qwen3.7-max via Qwen Code /review_

Author: Done in `da71ce481` — section renamed to "Why presubmit and cleanup live as `qwen review` subcommands". 已改名。

> - **`.qwen/review-tools.md`**: Let projects define custom lint/build/test commands. Precise, but requires users to learn a new config format and maintain it.

Author: **[Suggestion]** "custom lint/build/test commands" — with deterministic analysis removed, auto-discovery no longer covers lint tools. Only Agent 7 discovers build/test commands from CI config. Consider changing to "custom build/test commands" here and in the Decision paragraph below.

_— qwen3.7-max via Qwen Code /review_

Author: Done in `da71ce481` — changed to "custom build/test commands". 已改。

| `.qwen/review-tools.md` for custom tool config               | Requires users to learn a new format. Auto-discovery from CI config achieves the same result with zero user effort.       |
| Use fast model for verification/reverse audit                | User requirement: quality first. Fast models may miss subtle issues.                                                      |

> | Reduce to 2 agents (like Gemini)                             | Loses dimensional focus. Gemini compensates with deterministic tasks; we already have those AND want higher LLM coverage. |

Author: **[Suggestion]** "we already have those" (referring to deterministic tasks) is now factually incorrect — this PR removes the deterministic analysis step (Step 3), and only Agent 7's build/test remains as deterministic work. Consider updating to:

```suggestion
| Reduce to 2 agents (like Gemini)                             | Loses dimensional focus. We retain build/test (Agent 7) and want higher LLM coverage. |
```

_— qwen3.7-max via Qwen Code /review_

Author: Done in `da71ce481` — the cell now reads "We retain build/test (Agent 7) and want higher LLM coverage". 已修正。
